### PR TITLE
Addressing python error bug#20077

### LINKED ISSUE #20077: Code formatter path matching
**Title:** Code formatter path matching

**Description**:

The code formatter performs unexpected behavior when provided paths that either do not start with `./` or contain a trailing `/` due to lack of path normalization. When the formatter is run with arguments that do not meet these requirements, the formatter attempts to run on files that are explicitly excluded. 

**Repro steps:**

Navigate to the root of the envoy project and run either of the following commands:

```bash
# Incorrect behavior due to trailing slash
./tools/code_format/check_format.py fix './source/server/admin/'
# Incorrect behavior due to lack of leading ./
./tools/code_format/check_format.py fix 'source/server/admin'
```

Both commands output `ERROR` messages regarding the use of `std::regex` in files excluded in the script by `STD_REGEX_ALLOWLIST`. Instead the formatter should interpret the paths given and put them into a standard form before proceeding to check for matches against excluded files.

**Sample Command & Output:**

```
$ ./tools/code_format/check_format.py fix 'source/server/admin'

ERROR: From source/server/admin/utils.h
ERROR: source/server/admin/utils.h:22: Don't use std::regex in code that handles untrusted input. Use RegexMatcher
ERROR: From source/server/admin/stats_handler.cc
ERROR: source/server/admin/stats_handler.cc:83: Don't use std::regex in code that handles untrusted input. Use RegexMatcher
ERROR: source/server/admin/stats_handler.cc:143: Don't use std::regex in code that handles untrusted input. Use RegexMatcher
ERROR: source/server/admin/stats_handler.cc:177: Don't use std::regex in code that handles untrusted input. Use RegexMatcher
ERROR: source/server/admin/stats_handler.cc:203: Don't use std::regex in code that handles untrusted input. Use RegexMatcher
ERROR: From source/server/admin/utils.cc
ERROR: source/server/admin/utils.cc:43: Don't use std::regex in code that handles untrusted input. Use RegexMatcher
ERROR: source/server/admin/utils.cc:47: Don't use std::regex in code that handles untrusted input. Use RegexMatcher
ERROR: source/server/admin/utils.cc:49: Don't use std::regex in code that handles untrusted input. Use RegexMatcher
ERROR: From source/server/admin/prometheus_stats.cc
ERROR: source/server/admin/prometheus_stats.cc:52: Don't use std::regex in code that handles untrusted input. Use RegexMatcher
ERROR: source/server/admin/prometheus_stats.cc:54: Don't use std::regex in code that handles untrusted input. Use RegexMatcher
ERROR: source/server/admin/prometheus_stats.cc:83: Don't use std::regex in code that handles untrusted input. Use RegexMatcher
ERROR: source/server/admin/prometheus_stats.cc:259: Don't use std::regex in code that handles untrusted input. Use RegexMatcher
ERROR: From source/server/admin/prometheus_stats.h
ERROR: source/server/admin/prometheus_stats.h:30: Don't use std::regex in code that handles untrusted input. Use RegexMatcher
ERROR: From source/server/admin/stats_handler.h
ERROR: source/server/admin/stats_handler.h:53: Don't use std::regex in code that handles untrusted input. Use RegexMatcher
ERROR: source/server/admin/stats_handler.h:55: Don't use std::regex in code that handles untrusted input. Use RegexMatcher
ERROR: source/server/admin/stats_handler.h:63: Don't use std::regex in code that handles untrusted input. Use RegexMatcher
ERROR: source/server/admin/stats_handler.h:69: Don't use std::regex in code that handles untrusted input. Use RegexMatcher
ERROR: check format failed. run 'tools/code_format/check_format.py fix'
```

### PR DESCRIPTION
**Pull request by ankatare**
Signed-off-by: Abhay Narayan Katare <abhay.katare@india.nec.com>

Commit Message: Addressing Python error 
Additional Description: Code formatter path matching Issue in check_format.py file.
Risk Level: Low
Testing: Unit
Optional Fixes #20077

### CODE REVIEW

**Review by phlax — COMMENTED** [review]
thanks for this @ankatare 

couple of nits and we should be gtm

File: tools/code_format/check_format.py [hunk 1]
```diff
@@ -809,7 +810,10 @@ def check_source_line(self, line, file_path, report_error):
                 "other sinks can add the suffix automatically on flush should they prefer to do so."
             )
 
-        if not self.allow_listed_for_std_regex(file_path) and "std::regex" in line:
+        normalized_target_path = file_path
+        if not normalized_target_path.startswith("./"):
+            normalized_target_path = "./" + normalized_target_path
```
[the remark this answers is not in the document]
Comment (ankatare) [reply tools/code_format/check_format.py]: Thanks @phlax for your comments.. Actually this was the Only code changes i did.. remaining codes are unwanted due to unclean behabiour of my local branch.  Incorporated your suggetions ... Thanks

File: tools/code_format/check_format.py [hunk 2]
```diff
@@ -572,7 +573,7 @@ def has_cond_var_wait_for(self, line):
             return False
         preceding = line[0:wait_for]
         if preceding.endswith("time_system") or preceding.endswith("timeSystem()") or \
-          preceding.endswith("time_system_"):
+                preceding.endswith("time_system_"):
```
[the remark this answers is not in the document]
Comment (ankatare) [reply tools/code_format/check_format.py:576]: Yeah, this file needs lots of work on that. Agree

**Comment by ankatare** [comment]
@phlax .. Check Now... if it is go to merge ...

**Comment by phlax** [comment]
@ankatare is it possible to move the `# yapf enable` further up so the linting issues you fixed remain so ?

**Comment by ankatare** [comment]
@phlax Sure, let me try as i have bad environemt today :(

**Comment by phlax** [comment]
just the `yapf: enable` needs to be moved - so that less of the file is excluded from yapf checks

i would put it directly after `PROTOBUF_TYPE_ERRORS` 

afaict that was why it was added initially

not sure if it works inline or needs to be on a separate line

**Comment by ankatare** [comment]
> just the `yapf: enable` needs to be moved - so that less of the file is excluded from yapf checks
> 
> i would put it directly after `PROTOBUF_TYPE_ERRORS`
> 
> afaict that was why it was added initially
> 
> not sure if it works inline or needs to be on a separate line

@phlax Done.. Please check...

**Review by phlax — APPROVED** [review]
lgtm, thanks @ankatare

**Pull request merged**